### ISSUE TITLE
Do you want this features? Better search, resizable workspaces, etc

### ISSUE THREAD

**Issue by avently** [linuxdeepin/deepin-terminal#43]
Hello. I developed some usefull things for myself. I can make PR here if you want all/some of the features:

- (1) better search. Now after you pressed hotkey for search Ctrl+Shift+F and then pressed Enter or next/previous button in search panel the search will not happens. So you need to press something again in a search box to make the search working. I fixed it.
- (2) custom (user-provided) themes support located in ~/.config/deepin/deepin-terminal/themes. The same syntax but in another directory.
- (3) tabbar can be located at the bottom of the screen for tiling WMs. For tiling WMs "close" button in the top right corner is unneeded. So we can place tabbar at the bottom of the screen. I can make it customsizable.
- (4) Gtk.Menu that replace Deepin Menu dependency and have a liiiitle bit different look and feel with different themes.
- (5) refactored CMakeLists. Now you have to add every new *.vala file name to the CMakeLists file. It's not usefull. I did universal CMakeLists. Just take a look on it.
- (6) resizable workspaces. You can use Ctrl+Alt+Left/Right/Top/Bottom buttons to resize workspaces in any direction.

Here is a link for my repo:
https://github.com/avently/terminal/commits/master

Just choose what you want to get and I'll make a PR with one commit per feature.

**Comment by manateelazycat** [comment]
Hi avently

Sorry for late reply, it's really busy recently for me.

Your patch looks awesome, below is my suggestions:

1) I can't understand issue, can you give me some step to reproduce issue before i merge your patch?
2) This patch will break old version's config file compatibility
3) This feature is good, i'm happy to apply it.
4) Many un-deepin user report this problem, please sent single PR, thanks
5) It's wonderful, plase make a PR
6) Awesome, please make a PR

Please make a PR with per feature, it's easier for code review.

Thank very much for your great work, i'm look forward it.

**Comment by manateelazycat** [comment]
> (1) better search. Now after you pressed hotkey for search Ctrl+Shift+F and then pressed Enter or next/previous button in search panel the search will not happens. So you need to press something again in a search box to make the search working. I fixed it.

This patch has merged.

> (2) custom (user-provided) themes support located in ~/.config/deepin/deepin-terminal/themes. The same syntax but in another directory.

This patch i need consider compatibility problem.

> (3) tabbar can be located at the bottom of the screen for tiling WMs. For tiling WMs "close" button in the top right corner is unneeded. So we can place tabbar at the bottom of the screen. I can make it customsizable.

It be great if user can customize this with additional option in ~/.config/deepin/deepin-terminal/config.conf 
Please make a PR,thanks!

> (4) Gtk.Menu that replace Deepin Menu dependency and have a liiiitle bit different look and feel with different themes.

I think this patch is not good enough since deepin-menu is standard module in deepin system.

Can you change patch that user can choose which menu through configure CMake ?

I want provide a option that user can use Gtk Menu if user not use deepin system, but i don't want remove deepin-menu from code, deepin user need it.

> (5) refactored CMakeLists. Now you have to add every new *.vala file name to the CMakeLists file. It's not usefull. I did universal CMakeLists. Just take a look on it.

This patch is wonderful, has merged, thanks!


> (6) resizable workspaces. You can use Ctrl+Alt+Left/Right/Top/Bottom buttons to resize workspaces in any direction.

Can you make a PR? 
I like this feature. ;)

**Comment by avently** [comment]
@manateelazycat 
>This patch i need consider compatibility problem.

What compatibility problem you mean?

**Comment by cleversonahum** [comment]
@avently did your change about resizable workspaces already merged on deepin terminal? I am looking for this option

**Comment by avently** [comment]
@cleversonahum nope. I didn't send the changes because there is no fate that it will be merged. Author of the terminal has its own vision and that vision is not fully described so I can only guess

**Comment by manateelazycat** [comment]
@avently @cleversonahum 

I have already left from deepin, more features you can @hualet, thank you for your patches.

**Comment by avently** [comment]
@manateelazycat sad news. Thank you for creating the best terminal ever!

**Comment by hualet** [comment]
@BLumia please check this when you're available.
